include/posix/sys/types.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __POSIX_TYPES_H__
+#define __POSIX_TYPES_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include_next <sys/types.h>
+
+#ifdef CONFIG_PTHREAD_IPC
+#include <kernel.h>
+
+/* Thread attributes */
+typedef struct pthread_attr_t {
+	int priority;
+	void *stack;
+	size_t stacksize;
+	u32_t flags;
+	u32_t delayedstart;
+	u32_t schedpolicy;
+	s32_t detachstate;
+	u32_t initialized;
+} pthread_attr_t;
+
+typedef void *pthread_t;
+
+/* Mutex */
+typedef struct pthread_mutex {
+	struct k_sem *sem;
+} pthread_mutex_t;
+
+typedef struct pthread_mutexattr {
```

Comment:
so make this an empty struct instead of wasting 4 bytes on "unused"

Reply:
@andyross  implementation. We can take this as a fix later.